src/klein/resource.py

```diff
@@ -75,7 +75,10 @@ def _extractURLparts(request):
     @rtype: L{tuple} of L{unicode}, L{unicode}, L{int}, L{unicode}, L{unicode}
     """
     server_name = request.getRequestHostname()
-    server_port = request.getHost().port
+    if hasattr(request.getHost(), 'port'):
+        server_port = request.getHost().port
+    else:
+        server_port = 0
```

Comment:
But if there is no value set for `server_port`, what should be returned in place of `server_port` without breaking the method everywhere it is used?